Adds updated doc material for the 4.1 release of Kibana

palecur opened the pull request:
Closes #3733

Updates K4 docs for 4.1 features. Notable changes include bubble charts from line charts, new aggregations (Percentile Rank and dedicated range aggs for dates and IPv4 addresses), UI changes (Options tab for visualizations).

tbragin commented:
Based on a cursory review, looks like great progress! 

A couple of brief comments so far:
* field formatters - could we include key examples for each format type?
* import/export options in Settings >> Advanced don’t seem to be documented
* line chart- "current time marker" view option not documented
* discover -screenshots outdates, since we moved the index selector (no more wheel)
* tile map - where do we describe the new geo bounding box option? I may have missed it

rashidkpc commented:
I don't see anything about filter pinning either. It would be nice to have examples of where its useful, eg, creating a filter on the dashboard, and heading back to discover with it.

rashidkpc commented on docs/line.asciidoc:
```diff
@@ -27,9 +27,27 @@ NOTE: In Elasticsearch releases 1.4.3 and later, this functionality requires you
 
+Select the *Options* tab to change the following aspects of the chart:
+
+*Y-Axis Scale*:: You can select *linear*, *log*, or *square root* scales for the chart's Y axis.
```
Can you give examples of where each of these might be useful? A bit of material: http://www.forbes.com/sites/naomirobbins/2012/01/19/when-should-i-use-logarithmic-scales-in-my-charts-and-graphs/

palecur replied:
Added this section to the discussion of different axis scales: "You can use a log 
scale to display data that varies exponentially, such as a compounding interest chart, or a square root scale to regularize the display of data sets with variabilities that are themselves highly variable. This kind of data, where the variability is itself variable over the domain being examined, is known as _heteroscedastic_ data. For example, if a data set of height versus weight has a relatively narrow range of variability at the short end of height, but a wider range at the taller end, the data set is heteroscedastic."

rashidkpc commented on docs/line.asciidoc:
```diff
@@ -27,9 +27,27 @@ NOTE: In Elasticsearch releases 1.4.3 and later, this functionality requires you
+Select the *Options* tab to change the following aspects of the chart:
+
+*Y-Axis Scale*:: You can select *linear*, *log*, or *square root* scales for the chart's Y axis.
+*Smooth Lines*:: Check this box to curve the line from point to point.
```
Its worth pointing out to the user some of the downsides with curved lines: http://vizwiz.blogspot.com/2011/12/when-you-use-smoothed-line-chart-your.html

palecur replied:
Added the sentence " Bear in mind that smoothed lines necessarily affect the representation of your data and create a potential for ambiguity."

rashidkpc commented on docs/line.asciidoc:
```diff
@@ -27,9 +27,27 @@ NOTE: In Elasticsearch releases 1.4.3 and later, this functionality requires you
+
+*Y-Axis Scale*:: You can select *linear*, *log*, or *square root* scales for the chart's Y axis.
+*Smooth Lines*:: Check this box to curve the line from point to point.
+*Show Connecting Lines*:: Check this box to draw lines between the points on the chart.
```
If you turn these off and use the new Dot Size metric you can create a bubble chart!

rashidkpc commented on docs/line.asciidoc:
```diff
@@ -27,9 +27,27 @@ NOTE: In Elasticsearch releases 1.4.3 and later, this functionality requires you
\ No newline at end of file
+*Scale Y-Axis to Data Bounds*:: The default Y-axis bounds are zero and the maximum value returned in the data. Check 
+this box to change both upper and lower bounds to match the values returned in the data.
+
```
Check out the y-min/y-max pull, you'll need to add stuff around it to this list: https://github.com/elastic/kibana/pull/3464

palecur replied:
Added the following to handle this and @tbragin's comment re: current time marker. 
*Current time marker*:: For charts of time-series data, check this box to draw a red line on the current time.
*Set Y-Axis Extents*:: Check this box and enter values in the *y-max* and *y-min* fields to set the Y axis to specific values.

rashidkpc commented on docs/tilemap.asciidoc:
```diff
@@ -25,7 +25,10 @@ Before you choose a buckets aggregation, specify if you are splitting the chart
+drop-down. The *Precision* slider determines the granularity of the results displayed on the map. See the documentation 
+for the {ref}/search-aggregations-bucket-geohashgrid-aggregation.html#_cell_dimensions_at_the_equator[geohash grid] 
+aggregation for details on the area specified by each precision level. As of the 4.1 release, Kibana supports a maximum 
+geohash length of 7.
```
You should mention that higher precisions use more memory both in the browser and in elasticsearch, also we recently merged the auto precision pull, so you should mention that is checked by default

palecur replied:
Added this sentence:

"NOTE: Higher precisions increase memory usage for the browser displaying Kibana as well as for the underlying Elasticsearch cluster."

Also added 

"NOTE: By default, the *Change precision on map zoom* box is checked. Uncheck the box to disable this behavior."

after the Geohash entry. Question: Why is this option here, and not in Options with the other ones?

rashidkpc commented on docs/tilemap.asciidoc:
```diff
@@ -68,8 +80,19 @@ NOTE: In Elasticsearch releases 1.4.3 and later, this functionality requires you
\ No newline at end of file
+*Shaded Geohash Grid*:: Displays the rectangular cells of the geohash grid instead of circular markers, with different 
+shades based on the metric aggregation's value.
+*Desaturate map tiles*:: Desaturate the map's color in order to make the markers stand out more clearly.
```
Heat maps are going into this release, will need docs on those. Lots of new options here. https://github.com/elastic/kibana/pull/3830

palecur replied:
Details added in latest update to this PR.

rashidkpc commented on docs/visualize.asciidoc:
```diff
@@ -91,7 +91,7 @@ Use the aggregation builder on the left of the page to configure the {ref}/searc
 {ref}/search-aggregations.html[Elasticsearch aggregations reference].
 
-In bar or line chart visualizations, use _metrics_ for the y-axis and _buckets_ are used for the x-axis, segment bar 
+Bar or line chart visualizations use _metrics_ for the y-axis and _buckets_ are used for the x-axis, segment bar 
```
bar/line/area

rashidkpc commented:
A couple first impression notes on the tutorial

1. The user is asked to load the Shakespeare data set, but nothing is done with it after that
2. The accounts data appears to be *really* random, so there's nothing to really find in it. Might want to find a better set here. It also has some geo-ish data, but its all address based so the user can't do anything with maps. Maybe ask sales eng if they have anything?

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+=== Defining Your Index Patterns
+
+Each set of data loaded to Elasticsearch has an https://www.elastic.co/guide/en/kibana/current/settings.html#settings-create-pattern[index pattern]. In the previous section, the Shakespeare data set has an index named `shakespeare`, and the accounts 
+data set has an index named `bank`. An _index pattern_ is a regular expression that can 
```
Index patterns can not use regular expressions. Its a wildcard pattern at best.

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+Beneath the search box, the current index pattern is displayed in a drop-down. You can change the index pattern by 
+selecting a different pattern from the drop-down selector.
+
+Try selecting the `ba*` index pattern and putting the following search into the search box:
```
A bit more on query syntax would be good, and maybe a short plain-english explanation of the query.

palecur replied:
Added 
"You can construct searches by using the field names and the values you're interested in. With numeric fields you can use comparison operators such as greater than (>), less than (<), or equals (=). You can link elements with the logical operators AND, OR, and NOT, all in uppercase."
before the example, and
"This search returns all account numbers between zero and 99 with balances in excess of 47,500."
after the example.

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+
+The visualization tools available on the *Visualize* tab enable you to display aspects of your data sets in several 
+different ways. Visualizations depend on Elasticsearch {ref}/search-aggregations.html[aggregations] in two different 
+types: _bucket_ aggregations and _metric_ aggregations. A bucket aggregation sorts your data according to criteria you 
```
Feels like metrics and buckets would be better explained further down, once they're properly in the visualization builder.

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+
+Click the green *Apply changes* to display the chart:
+
+image::images/tutorial-visualize-pie-2.png[]
```
This screenshot has the builder collapsed. Expanding it would allow the user to follow along easier

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+Click *Add sub-buckets* at the bottom, then select the *Terms* aggregation and the *age* field from the drop-downs. 
+Click the green *Apply changes* button to add an external ring with the new results.
+
+image::images/tutorial-visualize-pie-3.png[]
```
The chart and legend would be both more meaningful and cleaner if you used a histogram on age, say with an interval of 10, instead of terms.

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+Select *Terms* as the sub-aggregation, with *state* as the field. Leave the other elements at their default values and 
+click the green *Apply changes* button. Your chart should now look like this:
+
+image::images/tutorial-visualize-bar-2.png[]
```
This is the problem with "random" data, its so smooth. See how all the blocks are the same size?

rashidkpc commented on docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,224 @@
+}
+
+After downloading the data sets, load them into Elasticsearch with the following commands:
+
```
We should really have the user import a mapping. Mappings are so important in elasticsearch that learning about them later on is sort of harmful. See this step in the Kibana 3 tutorial: https://www.elastic.co/guide/en/kibana/3.0/import-some-data.html

palecur replied:
I've added information on mapping imports.

rashidkpc commented:
The flow is a bit off here. The previous section of the docs talks about setting up index patterns, but of course you can't do that without data, which you get in the tutorial section. It seems the tutorial should be the first thing you hit after a "What is kibana and why do I care?" section

w33ble commented:
Some general feedback:

- `images/NYCTA-Dashboard.jpg` should be better cropped (left and bottom edges removed)
- The Visualize section is huge - would it maybe make sense to break each of the vis types out into their own page or something?

tbragin commented:
My comments after the latest round of changes. 

**Add distribution package formats & service definitions** https://github.com/elastic/kibana/pull/3212
	=> No content. Should add information on the new available packaging and installation options to "Getting Kibana Up and Running". Right now we only describe the manual installation. 

**Collapsible sidebar** https://github.com/elastic/kibana/pull/3185
	=> No content. Should mention that in the Discover section and update the out-of-date screenshot. 

**Discover interval** https://github.com/elastic/kibana/pull/3290
	=> No content. Should describe how to change the default 

**Add columns from doc table** https://github.com/elastic/kibana/pull/3526
	=> No content. Should describe the process for doing that

**Move index pattern selector into index pattern header in discover** https://github.com/elastic/kibana/pull/3060
	=> No content. Should describe the process for doing that

**Add IPv4 Range Aggregation** https://github.com/elastic/kibana/pull/3171
        => IPv4 Range URL is a broken link: http://www.elastic.co/guide/en/elasticsearch/reference/currentsearch-aggregations-bucket-iprange-aggregation.html
       => Date Range and IPv4 Range options are not present in Tilemap, but we document them as if they are. 

**vis editor refactor** https://github.com/elastic/kibana/pull/3616
    => Visualize screenshot needs to be updated to show vis editor restyle (we should especially point to where options moved into, since they are much less discoverable now)

**Add ability to select y-axis min and max values** https://github.com/elastic/kibana/pull/3464
	=> "Set Y-Axis Extents" not described

**Adding feature to remove circles when option to remove circles selected** https://github.com/elastic/kibana/pull/3197
	=> My understanding is that circles are shown by default. So we should document this as "uncheck the box to remove the circles…"

**Tilemap PRs**
	=> "Navigating the Map" Formatting is off… bullet points do not show up as expected
	=> We should consider including screenshots of the buttons inline with instructions in more places, because it’s not clear what the "Fit Data Bounds" button is, unless you mouse over

**Add custom date histogram interval option**  https://github.com/elastic/kibana/pull/3237
	=> No content. Should describe the "Custom" option in the "Date Histogram" blurb

**Add a pause button to the header when auto-refresh is enabled**
https://github.com/elastic/kibana/pull/3800
	=> No content on the pause button
	=> Also, currently time picker is only described in Discover, however this functionality, including auto-refresh is available in Discover, Visualize, and Dashboard. Auto-refresh is very important in the Dashboard page. 

**Add option to save time with dashboard**  https://github.com/elastic/kibana/pull/3164
	=> "Store time with dashboard" checkbox not described.

**Settings refactor**
	- We discussed a refactor to make "Managing Saved Searches, Visualizations, and Dashboards" easier to find: break up the page into "Indices", "Advanced", and "Objects" and link to them at the top, like we do in visualize. 

**Import/export** https://github.com/elastic/kibana/pull/3573
	=> Formatting of the paragraph off - bullet points do not show up 

**Closes #2471 - Add logging to a file** https://github.com/elastic/kibana/pull/3201
	=> Do we describe startup options anywhere? We should… are there others?

tbragin commented:
On the tutorial: Link to the logstash dataset does not seem to be present in the latest version
![screen shot 2015-05-20 at 11 38 17 am](https://cloud.githubusercontent.com/assets/933397/7733554/bd818c74-fee4-11e4-911c-77877d470361.png)

palecur commented:
The logstash data set isn't hosted yet; I've talked to Drew about it and he'll take care of it. I'll update it as soon as I know the URL.

palecur commented:
Pull #3212: Added this section to setup.asciidoc

On Unix, you can instead run the package manager suited for your distribution:

* Red Hat, Centos: `yum install kibana`
* Debian: `apt-get install kibana`

After installing, run Kibana from the install directory: `bin/kibana` (Linux/MacOSX) or `bin\kibana.bat` (Windows).

I'll need information from @jordansissel about where those packages' default install directories are so I can add that to the section, but it's not much more involved than that.

Pull #3185: done.

Pull #3290: Screenshot updated, text added: The histogram lists the time range you're currently exploring, as well as the intervals that range is currently using. 
To change the intervals, click the link and select an interval from the drop-down. The default behavior automatically sets an interval based on the time range.

Pull #3464: Added to area chart, already present in line chart

Pull #3526: Added ". To toggle a particular field's column in the Documents table, click the 
image:images/add-column-button.png[Add Column] *Toggle column in table* button." and a representative screenshot.

Pull #3060: Current doc says to click the name of the index pattern to change it; this behavior still works, since the directional indicator is in the same bar as the pattern name, and clicking anywhere on the bar unfolds the list of index patterns. Screenshot updated to reflect new visual style.

Pull #3171: Fixed bad link. IPv4 and date range aggs are definitely selectable and present as sub-aggregations.

Pull #3616: Updated screenshot and text referring to Apply changes button.

More to come later tonight with PR update and rebuild of sample doc for further inspection

palecur commented:
Pull #3197: Typically we've just been documenting what the box does when it's checked, not altering our phrasing by whether or not it's default-checked.

Tilemap: Format fixed, images for buttons added. +/- zoom buttons added and documented.

#3237: The following language added to all discussions of the Date Histogram agg (x-axis-aggs, tilemap, pie, and datatable .asciidoc):
"You can also specify a custom interval frame by selecting *Custom* as the interval and 
specifying a number and a time unit in the text field. Custom interval time units are *s* for seconds, *m* for minutes, *h* for hours, *d* for days, *w* for weeks, and *y* for years. Different units support different levels of precision, down to one second."

#3800: Autorefresh added to discussion of the time picker, and is now present in Dashboard and Visualize as well.

#3164: Added the following text:
By default, dashboards store the time period specified in the time filter when you save a dashboard. To disable this behavior, clear the *Store time with dashboard* box before clicking the *Save* button.

Settings chunking: Done

#3201: Yes, these are all in Settings. I've added a line for log_file right after pid_file.

PR updated, new build up for inspection as usual.

tbragin commented:
Perhaps an item for a later date, but for the Kibana server properties table, do we also want to document the command-line options that go along with each option? 

    -h, --help                 output usage information
    -V, --version              output the version number
    -e, --elasticsearch <uri>  Elasticsearch instance
    -c, --config <path>        Path to the config file
    -p, --port <port>          The port to bind to
    -q, --quiet                Turns off logging
    -H, --host <host>          The host to bind to
    -l, --log-file <path>    The file to log to
    --plugins <path>           Path to scan for plugins

tbragin commented:
Link to IPv4 Range aggregation still missing a slash between "current" and "search": 

http://www.elastic.co/guide/en/elasticsearch/reference/currentsearch-aggregations-bucket-iprange-aggregation.html

tbragin commented:
On pull https://github.com/elastic/kibana/pull/3212, agreed that need info from @jordansissel and/or @rashidkpc on whether the kibana RPM and DEB packages are going to be uploaded to the respective repos, so the following installation commands would work at the time when we release these docs, as right now they don't seem to:
* Red Hat, Centos: yum install kibana
* Debian: apt-get install kibana

tbragin commented:
"What's new in Kibana 4.1" content looks good to me. I would probably not use bold for the pull numbers, but it's only a preference. 

We should modify the note in the intro that refers to which product version this documentation is for, for instance:

This guide describes how to use Kibana 4**.1**. For information about what’s new in Kibana 4**.1**, see What’s New in Kibana 4.1. **For earlier versions of Kibana 4 go here [include link].** For information about Kibana 3, see the Kibana 3 User Guide.
